pkg/operator/controller/controller_cluster_role_test.go

```diff
@@ -0,0 +1,49 @@
+package controller
+
+import (
+	"testing"
+
+	"github.com/openshift/cluster-dns-operator/pkg/manifests"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+func TestDNSClusterRoleChanged(t *testing.T) {
+	testCases := []struct {
+		description string
+		mutate      func(*rbacv1.ClusterRole)
+		expect      bool
+	}{
+		{
+			description: "if nothing changes",
+			mutate:      func(_ *rbacv1.ClusterRole) {},
+			expect:      false,
+		},
```

Comment:
Maybe add a test case where something that should be ignored changes, such as adding an annotation or label.

Reply:
done.